fineract-provider/src/test/java/org/apache/fineract/cob/loan/CheckLoanRepaymentOverdueBusinessStepTest.java

```diff
@@ -175,4 +184,27 @@ public void givenLoanWithInstallmentOverdueAfterConfiguredDaysInLoanProductWhenS
         assertEquals(repaymentInstallment, loanPayloadForEvent);
         assertEquals(processedLoan, loanForProcessing);
     }
+
+    @Test
+    public void givenNonDisbursedLoanWhenStepExecutionThenNoBusinessEventIsRaised() {
```

Comment:
I am not fully understand this test case, but please be sure to cover the following:
GIVEN Loan is SUBMITTED or APPROVED and outstanding amount is greater than ZERO and one of the instalment is OVERDUE
WHEN job got executed
THEN no overdue event

GIVEN Loan is ACTIVE, outstanding amount is ZERO, one of the installment is due date is matching with the actual date
WHEN job got executed
THEN no overdue event triggered

Reply:
Done! Test cases created:

`givenActiveLoanWithZeroOutstandingWhenStepExecutionThenNoBusinessEventIsRaised`
`givenActiveLoanWithNonZeroOutstandingWhenStepExecutionThenBusinessEventIsRaised`